packages/ui/src/memberships/hooks/useMembers.ts

```diff
@@ -12,32 +14,46 @@ export interface MemberListOrder {
 
 export const DefaultMemberListOrder: MemberListOrder = { sortBy: 'id', isDescending: false }
 
+export const MEMBERS_PER_PAGE = 5
+
 interface UseMemberProps {
   order: MemberListOrder
   filter: MemberListFilter
+  page?: number
 }
 interface UseMembers {
   isLoading: boolean
   members: Member[]
+  pageCount?: number
 }
 
-export const useMembers = ({ order, filter }: UseMemberProps): UseMembers => {
+export const useMembers = ({ order, filter, page = 1 }: UseMemberProps): UseMembers => {
   const variables = {
-    where: { ...filterToGqlInput(filter) },
+    first: page * MEMBERS_PER_PAGE,
+    last: page > 1 ? MEMBERS_PER_PAGE : undefined,
```

Comment:
I see "[Including a value for both first and last is strongly discouraged](https://relay.dev/graphql/connections.htm#note-a97ec)", but because the current pagination allows to click on a arbitrary page number, if I'm on page 1 and I click to page 3 (and there is 5 items/page), should it send: { after: "last-item-of-p1", first: 10 } ? This approach would load a lot of data if someone goes straight to the last page. Or maybe in case there is only 3 pages: { last: 5 } ? This way it limits the loaded items when clicking on the last page.

Or should we change the pagination component to only show next/previous. Or is there something I'm missing ?

Reply:
That's a good question :) I'll see tomorrow how it should be done.